src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -255,7 +255,7 @@ private async Task ShowProgressUIAsync()
         public async Task<bool> UIRestorePackagesAsync(CancellationToken token)
         {
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-            ShowProgressUI();
+            await ShowProgressUIAsync();
```

Comment:
Similar to the GetPropertyValue changes, this actually just adds an additional state machine to handle the async side. 

When you are already on the UI thread, and you are calling another method that fully runs on the UI thread, an async method would always perform worse.